manifests/server.pp

```diff
@@ -89,79 +91,80 @@
-  Boolean $manage_postgresql_conf_perms            = $postgresql::params::manage_postgresql_conf_perms,
-  Boolean $manage_selinux                          = $postgresql::params::manage_selinux,
-  $module_workdir                                  = $postgresql::params::module_workdir,
-
-  $manage_datadir                                  = $postgresql::params::manage_datadir,
-  $manage_logdir                                   = $postgresql::params::manage_logdir,
-  $manage_xlogdir                                  = $postgresql::params::manage_xlogdir,
-  $password_encryption                             = $postgresql::params::password_encryption,
-  $extra_systemd_config                            = $postgresql::params::extra_systemd_config,
-
-  Hash[String, Hash] $roles              = {},
-  Hash[String, Any] $config_entries      = {},
-  Postgresql::Pg_hba_rules $pg_hba_rules = {},
-
-  Boolean $backup_enable = $postgresql::params::backup_enable,
-  Hash $backup_options = {},
-  Enum['pg_dump'] $backup_provider = $postgresql::params::backup_provider,
+  String[1]                                    $package_name                 = $postgresql::params::server_package_name,
+  Enum['present', 'absent', 'latest']          $package_ensure               = $postgresql::params::package_ensure,
+
+  Optional[String[1]]                          $plperl_package_name          = $postgresql::params::plperl_package_name,
+  Optional[String[1]]                          $plpython_package_name        = $postgresql::params::plpython_package_name,
+
+  Variant[Enum['running', 'stopped'], Boolean] $service_ensure               = $postgresql::params::service_ensure,
+  Boolean                                      $service_enable               = $postgresql::params::service_enable,
+  Boolean                                      $service_manage               = $postgresql::params::service_manage,
+  String[1]                                    $service_name                 = $postgresql::params::service_name,
+  Boolean                                      $service_restart_on_change    = $postgresql::params::service_restart_on_change,
+  Optional[String[1]]                          $service_provider             = $postgresql::params::service_provider,
+  String[1]                                    $service_reload               = $postgresql::params::service_reload,
+  String[1]                                    $service_status               = $postgresql::params::service_status,
+  String[1]                                    $default_database             = $postgresql::params::default_database,
+  Hash                                         $default_connect_settings     = $postgresql::globals::default_connect_settings,
+  Optional[String[1]]                          $listen_addresses             = $postgresql::params::listen_addresses,
+  Variant[String[1], Integer]                  $port                         = $postgresql::params::port,
```

Comment:
@ekohl mentioned that in #1396 `Variant[String[1], Stdlib::Port] $port` would probably be the better fit

Reply:
And if this is a breaking change, I'd even suggest to tighten it to just `Stdlib::Port`.

Reply:
We are trying to avoid turning this series of syntax updates into breaking changes. But thanks for the suggestion, will implement as soon as I can.

Reply:
That's fair, but you can safely use `Stdlib::Port` instead of `Integer`. Using an integer outside of the valid port range would result in the application failing to start up anyway so I'd not consider it breaking.

Reply:
Done